Sources/AppStoreConnectCLI/Commands/TestFlight/BetaTesters/ListBetaTestersCommand.swift

```diff
@@ -0,0 +1,97 @@
+    var bundleId: String?
+
+    @Option(help: "The name of the beta group")
+    var betaGroupName: String?
+
+    @Option(help: "The ID of one build of an application")
+    var buildId: String?
+
+    private enum ListStrategy {
+        case all
+        case listByApp(bundleId: String)
+        case listByGroup(betaGroupName: String)
+        case listByBuild(buildId: String)
+
+        typealias ListOptions = (bundleId: String?, betaGroupName: String?, buildId: String?)
+
+        init(options: ListOptions) {
+            switch (options.bundleId, options.betaGroupName, options.buildId) {
+                case let(.some(bundleId), _, _) where !bundleId.isEmpty:
+                    self = .listByApp(bundleId: bundleId)
+                case let(_, .some(betaGroupName), _) where !betaGroupName.isEmpty:
+                    self = .listByGroup(betaGroupName: betaGroupName)
+                case let(_, _, .some(buildId)) where !buildId.isEmpty:
+                    self = .listByBuild(buildId: buildId)
+                case (_, _, _):
+                    self = .all
+            }
+        }
+    }
+
+    func run() throws {
+        let api = try makeClient()
+
+        let request: AnyPublisher<BetaTestersResponse, Error>
+
+        let includes = [ListBetaTesters.Include.apps, ListBetaTesters.Include.betaGroups]
+
+        switch ListStrategy(options: (bundleId, betaGroupName, buildId)) {
+            case .all:
+                request = api
+                    .request(APIEndpoint.betaTesters(include: includes))
+                    .eraseToAnyPublisher()
+
+            case .listByApp(let bundleId):
+                request = api
+                    .getAppResourceIdsFrom(bundleIds: [bundleId])
+                    .flatMap {
+                        api.request(APIEndpoint.betaTesters(
+                            filter: [ListBetaTesters.Filter.apps($0)],
+                            include: includes
+                        ))
+                    }
+                    .eraseToAnyPublisher()
+
+            case .listByGroup(let betaGroupName):
+                request = try api.betaGroupIdentifier(matching: betaGroupName)
+                    .flatMap {
+                        api.request(APIEndpoint.betaTesters(
+                            filter: [ListBetaTesters.Filter.betaGroups([$0])],
+                            include: includes
+                        ))
+                    }
+                    .eraseToAnyPublisher()
+            
+            case .listByBuild(let buildId):
+                request = api.request(APIEndpoint.betaTesters(
+                        filter: [ListBetaTesters.Filter.builds([buildId])],
+                        include: includes
+                    ))
+                    .eraseToAnyPublisher()
+        }
+
+        _ = request
+            .map(\.data)
+            .flatMap { api.fromAPIBetaTesters(betaTesters: $0) }
```

Comment:
This flatMap is going to create a whole load of redundant requests.  Which is probably why this command can take so long now if you have a bunch of beta testers. With an account with 50 beta testers it takes nearly 30seconds to run.

Reply:
Yes, because the response of getting beta tester will only return Ids of Apps and Ids of Groups, in order to get the name and bundle Id of the group and App, I use the Id to fetch Groups/Apps respectively.